activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb

```diff
@@ -35,9 +36,11 @@ module ConnectionAdapters
     class Mysql2Adapter < AbstractMysqlAdapter
       ADAPTER_NAME = 'Mysql2'.freeze
 
+      include MySQL::DatabaseStatements
+
       def initialize(connection, logger, connection_options, config)
         super
-        @prepared_statements = false
+        @prepared_statements = false unless config.key?(:prepared_statements)
```

Comment:
We should respect it. If users ask to enable it, it should be enabled, to disable it, it should be disabled. This implementation while keeping the default is enabling if you do `prepared_statements: false` in your config.